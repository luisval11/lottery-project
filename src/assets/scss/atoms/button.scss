@import "../abstracts/mixins";

button {
	background: linear-gradient(
		90deg,
		$color-button-01 0%,
		$color-button-02 100%
	);
	border-radius: 4px;
	box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.16);
	color: $color-ui-casino-01;
	font-size: 14px;
	line-height: 21px;
	text-align: center;
	@include CTA;
	min-height: 40px;
	padding: 7px 14px 7px 14px;
}
