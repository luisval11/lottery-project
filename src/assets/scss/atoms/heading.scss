@import "../abstracts/variables";

h4 {
	color: $color-ui-casino-02;
	font-feature-settings: "pnum" on, "lnum" on;
	font-size: 21px;
	@include CTA;
	line-height: 24px;
}

h5 {
	color: $color-ui-casino-02;
	font-feature-settings: "pnum" on, "lnum" on;		
	font-size: 17px;
	@include CTA;
	line-height: 21px;
	margin: 2px 0px;
}