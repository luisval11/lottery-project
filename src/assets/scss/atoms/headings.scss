@import "../abstracts/variables";

h5 {
    font-weight: 900;
    font-size: 17px;
    line-height: 21px;
    text-transform: uppercase;
    font-feature-settings: 'pnum' on, 'lnum' on;
    color: $color-ui-casino-02;
}

