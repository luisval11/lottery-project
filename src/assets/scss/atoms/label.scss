@import "../abstracts/variables";

label {
	color: $color-ui-casino-03;
	display: flex;
	align-items: center;
	font-feature-settings: "pnum" on, "lnum" on;
	font-size: 14px;
	font-weight: 500;
	line-height: 21px;
	& input[type="radio"] {
		margin-right: 15px;
	}
}
