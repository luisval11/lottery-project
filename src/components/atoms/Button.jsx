import { h } from "preact";

export const Button = ({ text, ...rest }) => {
	return <button {...rest}>{text}</button>;
};
