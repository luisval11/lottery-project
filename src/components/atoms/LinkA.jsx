import { h } from "preact";

export const LinkA = ({ href, ...rest }) => {
	return <a href={href} {...rest} />;
};
