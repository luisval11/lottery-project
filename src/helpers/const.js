export const URL_LOTTOLAND = "https://www.lottoland.co.uk";
export const SORT_OPTION_NAME = "name";
export const SORT_OPTION_EUR_STAKE_ASC = "stakeEurAsc";
export const SORT_OPTION_EUR_STAKE_DESC = "stakeEurDesc";
export const MAX_GAMES_PER_PAGE = 9;
export const SORT_OPTIONS = [
	{ name: SORT_OPTION_NAME, text: "Name (A-Z)" },
	{ name: SORT_OPTION_EUR_STAKE_ASC, text: "Stake (min to max)" },
	{ name: SORT_OPTION_EUR_STAKE_DESC, text: "Stake (max to min)" },
];
export const EXAMPLE_LISTITEM = {
	LuckyWizard: {
		name: "LuckyWizard",
		displayName: "Lucky Wizard",
		playURL: "/games/luckywizard/play",
		image: "/cms/5bab9d950eb3580fac83392e/Icon_320and250_LuckyWizard.jpg",
		detailURL: "/games/luckywizard",
		currencyData: {
			CHF: {
				noDesk: true,
				noMob: true,
			},
			EUR: {
				minimumStake: 0.1,
			},
			GBP: {
				minimumStake: 0.1,
			},
			SEK: { minimumStake: 1.0 },
			PLN: {},
			JPY: {},
			INR: {},
			NZD: { minimumStake: 0.1 },
			CAD: {},
			HUF: {},
		},
		volatility: ["High"],
		provider: ["RedTiger"],
	},
	GreekGods: {
		name: "GreekGods",
		displayName: "Greek Gods",
		playURL: "/games/greekgods/play",
		image: "/cms/5de91d2a91253e002efe72f7/icon-320and250_GreekGods.jpg",
		detailURL: "/games/greekgods",
		currencyData: {
			EUR: { minimumStake: 0.25 },
			GBP: { minimumStake: 0.25 },
			SEK: { minimumStake: 2.5 },
			PLN: {},
			BRL: {},
			JPY: {},
			NZD: { minimumStake: 0.25 },
			INR: {},
			HUF: {},
		},
		volatility: ["High"],
		provider: ["Pragmatic"],
	},
};
