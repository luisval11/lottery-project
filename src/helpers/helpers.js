import { SORT_OPTION_EUR_STAKE_ASC, SORT_OPTION_EUR_STAKE_DESC, SORT_OPTION_NAME, URL_LOTTOLAND } from "./const";

export const normalizeUrl = (endpoint) => {
	return URL_LOTTOLAND + endpoint;
};

export const isValidGame = (game) => {
	return (
		game.currencyData != undefined &&
		game.currencyData.EUR != undefined &&
		game.currencyData.EUR.minimumStake != undefined
	);
};

export const getItemsToShow = (sort, initialValues, initial, size) => {
	switch (sort) {
		case SORT_OPTION_NAME:
			return Object.values(initialValues)
				.filter((game) => isValidGame(game))
				.sort((gameA, gameB) => {
					return gameA.displayName.localeCompare(gameB.displayName);
				})
				.slice(initial, size);

		case SORT_OPTION_EUR_STAKE_ASC:
			return Object.values(initialValues)
				.filter((game) => isValidGame(game))
				.sort((gameA, gameB) => {
					return (
						gameA.currencyData.EUR.minimumStake -
						gameB.currencyData.EUR.minimumStake
					);
				})
				.slice(initial, size);

		case SORT_OPTION_EUR_STAKE_DESC:
			return Object.values(initialValues)
				.filter((game) => isValidGame(game))
				.sort((gameA, gameB) => {
					return (
						gameB.currencyData.EUR.minimumStake -
						gameA.currencyData.EUR.minimumStake
					);
				})
				.slice(initial, size);
	}
};
