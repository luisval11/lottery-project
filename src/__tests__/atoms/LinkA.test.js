import { h } from "preact";
import { mount } from "enzyme";
import { LinkA } from "../../components/atoms/LinkA";
import { URL_LOTTOLAND } from "../../helpers/const";

describe("Test on LinkA", () => {
	test("should match initial snapshot", () => {
		let wrapper = mount(
			<LinkA href={"#"}/>
		);
		expect(wrapper).toMatchSnapshot();
	});

	test("should have the value of const 'URL_LOTTOLAND' for href", () => {
		const hrefParam = URL_LOTTOLAND;
		let wrapper = mount(<LinkA href={hrefParam} />);
		const { href } = wrapper.find("a").props();
		expect(href).toBe(hrefParam);
	});

    test("should have a value 'Play' for text", () => {
		const textParam = "Play";
		let wrapper = mount(<LinkA text={textParam} />);
		expect(wrapper.find('a').text().trim()).toBe(textParam);
	});
});
