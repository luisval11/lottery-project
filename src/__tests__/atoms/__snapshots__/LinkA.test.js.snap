// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on LinkA should match initial snapshot 1`] = `ReactWrapper {}`;
